### DISCUSSION TITLE
premake5 not generating .sln file

### DISCUSSION THREAD

**Discussion by CrispNight** [premake/premake-core#2109] [Q&A] [answered]
I’m really new, so I apologize if I don’t get all the wording right for what I’m saying.
So, I have a C++ project in VS code and am trying to use premake5 to generate a .sln for VS 2022. From what I can tell it's just not generating a .sln or .vcxproj files. It's also not telling me why. 

When I type `premake5 vs2022` into the cmd I get back the following:

```
D:\Users\Bradford\source\repos\Test>premake5 vs2022    
Building configurations...
Running action 'vs2022'...
Done (26ms).
```
The only thing different from my premake5.lua file and the video I was trying to follow along with, [this one](https://www.youtube.com/watch?v=7sLeMVUo8Kg) , is the project name. Here it is:

```
workspace "Test"
    configurations { "Debug", "Release" }
    architecture "x64"

    project "HelloWorld"
        kind "ConsoleApp"
        language "C++"
        cppdialect "C++20"

        targetdir   "build/%{cfg.buildcfg}/bin"
        objdir      "build/%{cfg.buildcfg}/obj"

        location "./src"
        files { "%{prj.location}/**.h", "%{prj.location}/**.cpp" }

        filter "configurations:Debug"
            defines { "DEBUG" }
            symbols "On"
        filter {}

        filter "configurations:Release"
            defines { "NDEBUG" }
            optimize "On"
        filter {}
```

In all the videos I've seen the output to `premake5 vs2022` usually looks like this:
```
D:\Users\...\Test>premake5.exe vs2022 
Building configurations...
Running action 'vs2022'...
Generated Test.sln...
Generated src/HelloWorld.vcxproj...
Done (26ms).
```

I'm assuming its just not generating the files, I just don't know why. I've tried looking over the documentation for premake5 for hours and tweaking the premake5.lua file, but haven't figured it out. 

I aslo tried just copy and pasting the given example on the docs and got the same output.

**Comment by nickclark2016** [comment]
Which version of Premake are you running?

**Reply by nickclark2016** [reply]
I just tested with the 5.0.0-beta2 binary and cannot reproduce this issue. I recommend checking permissions in your folder.

**Reply by CrispNight** [reply]
It's version 5.0.0-beta2. I'm sorry, but what folder exactly and who should I be giving permissions to? I tried going into the properties of the project and src folder and giving Users and Authenticated Users full control. Still have the same results.

**Reply by nickclark2016** [reply]
Yeah, just wanted to make sure you had write permissions for the folder you were trying to run in.  Definitely weird, as I can't seem to reproduce the same issue you're seeing.

**Reply by nickclark2016** [reply]
If you try running a different generator, does it work?

**Reply by CrispNight** [reply]
I tried it with the vs2005-vs2022, codelite, gmake, and gmake2 actions. All to the same results. It's says 'Building configurations...' then running that action, then done. No files for any of them.

**Reply by nickclark2016** [reply, marked as answer]
Weird. Just as a sanity check, you did save the file, correct?

**Reply by CrispNight** [reply]
I verified all the files were saved. Ran it again, didn't work. Closed VS code, reopened VS code and ran it, and now it works. Generated a .sln and .vcxproj. Maybe it wasn't saved. Thank you so much for the help.
